### DISCUSSION TITLE
Add an entity option to limit the maximum volume of media players

### DISCUSSION THREAD

**Discussion by emontnemery** [home-assistant/architecture#1050] [Entity Models]
## Add an option to limit the maximum volume of media players

To facilitate this:
- A new entity option under the media_player domain is added
- A new property `volume_step_size` is added to the media player entity. The `volume_step_size` property has two purposes
  - Encourage integration authors to not implement custom `volume_up` or `volume_down` services
  - If the media player's `volume` + `volume_step_size` would exceed the limit set by the user, the `volume_up` service will not be called.

Media players which implement the `volume_up` or `volume_down` services are responsible for making sure the volume limit set by the user is not exceeded when the `volume_up` service is called, for example, a polling entity may need to do a state refresh before forwarding the call to the device. For a background to why this is suggested, please see the discussion in https://github.com/home-assistant/architecture/discussions/1012.

### Rationale

A maximum volume feature is already implemented in multiple media player integrations, both to improve the user experience when only the lower end of the volume range is useful, but more importantly to prevent damage when a high powered amplifier is connected to too small speakers.

**Comment by Drafteed** [comment]
> Media players which implement the volume_up or volume_down services are responsible for making sure the volume limit set by the user is not exceeded when the volume_up service is called, for example, a polling entity may need to do a state refresh before forwarding the call to the device. For a background to why this is suggested, please see the discussion in https://github.com/home-assistant/architecture/discussions/1012.

Sounds better than https://github.com/home-assistant/architecture/discussions/1012 👍
